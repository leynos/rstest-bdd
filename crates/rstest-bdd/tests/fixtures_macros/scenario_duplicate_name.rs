//! Compile-time fixture verifying that duplicate scenario titles require an
//! index selector.
use rstest_bdd_macros::{given, scenario, then, when};

#[given("a precondition")]
fn precondition() {}

#[when("an action occurs")]
fn action() {}

#[then("a result is produced")]
fn result() {}

#[scenario(path = "duplicate_titles.feature", name = "Shared title")]
fn duplicate_titles() {}

const _: &str = include_str!("duplicate_titles.feature");

fn main() {}
