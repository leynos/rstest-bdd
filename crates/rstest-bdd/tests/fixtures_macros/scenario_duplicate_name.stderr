error: found multiple scenarios named "Shared title"; use the `index` selector to disambiguate (matching indexes: 0, 1; lines: 3, 8)
  --> tests/fixtures_macros/scenario_duplicate_name.rs:14:54
   |
14 | #[scenario(path = "duplicate_titles.feature", name = "Shared title")]
   |                                                      ^^^^^^^^^^^^^^
